lib/curator.js

```diff
@@ -141,9 +141,23 @@ bedrock.events.on('bedrock-express.configure.routes', function(app) {
     '/tasks/credentials/compose-identity',
     relaxedAuthenticate, acceptableContent('urlencoded'),
     parseForm, parseJsonData, function(req, res, next) {
+    var credentialQuery = req.body.query;
+
+    /*
+     * pubicKey has the following properties
+     * type: array that always contains 'CryptographicKey' and can also contain
+     * ‘EmphemeralCryptographicKey’ or ‘NewCryptographicKey’
+     * id:, publicKeyPem:
+     */
+    var publicKey = req.body.publicKey;
+    if(publicKey.owner !== req.user.identity.id) {
+      // the identity referenced in the credential query does not correspond
+      // to the authenticated user.  Log-out the current user.
+      return res.redirect('/session/logout');
```

Comment:
We need a TODO here that says we need to queue the request for handling after logout. Maybe another TODO that says we could optimize by not redirecting to logout but instead making a call to clear the session and then redirecting to login w/some way of indicating that a request is queued.